---
layout: home
description: "Site home."
tags: home
---

I am a doctoral researcher at the [Dept. Algal Development and Evolution](https://www.bio.mpg.de/48867/algal-development-and-evolution-s) as well as the [Computational Biology Group](https://drostlab.com/) at the [Max Planck Institute for Biology, Tübingen](https://www.bio.mpg.de/).

---

## Bioinformatics
Contributions to open-source bioinformatic software applications.

[<img src="https://github.com/LotharukpongJS/LotharukpongJS.github.io/blob/main/images/orthologr_logo.png?raw=true" width="150">](https://drostlab.github.io/orthologr) [<img src="https://github.com/LotharukpongJS/LotharukpongJS.github.io/blob/main/images/myTAI_logo.png?raw=true" width="150">](https://drostlab.github.io/myTAI) [<img src="https://github.com/LotharukpongJS/LotharukpongJS.github.io/blob/main/images/phylomapr_logo.png?raw=true" width="150">](https://lotharukpongjs.github.io/phylomapr) [<img src="https://github.com/LotharukpongJS/LotharukpongJS.github.io/blob/main/images/GenEra_logo_dark.png?raw=true" width="150">](https://github.com/josuebarrera/GenEra)

[`orthologr`](https://drostlab.github.io/orthologr) provides a framework to perform large scale **comparative genomics** studies with R.

[`myTAI`](https://drostlab.github.io/myTAI) introduces procedures summarised under the term **evolutionary transcriptomics** to integrate gene age information into classical gene expression analysis in R.

[`phylomapr`](https://lotharukpongjs.github.io/phylomapr/) enables quick retrieval of precomputed **gene age maps** (phylomaps) in R.

[`GenEra`](https://github.com/josuebarrera/GenEra) estimates **gene-family founder events** (i.e., the age of the last common ancestor of protein-coding gene families) through the reimplementation of genomic phylostratigraphy at the tree of life scale.

## Biology
Outside of solitary confinement (and/or the dry lab office), we are surrounded by life. I love to learn about the diversity of living forms and the processes that shape their development and evolution. Here are some of my biological adventures.

Algal sample collection in St. Abbs, Scotland.

<img src="https://github.com/LotharukpongJS/LotharukpongJS.github.io/blob/main/images/PXL_20230821_151113739.jpg?raw=true" width="200" alt="The sea over St. Abbs."> <img src="https://github.com/LotharukpongJS/LotharukpongJS.github.io/blob/main/images/PXL_20230822_120843250.jpg?raw=true?raw=true" width="200" alt="Tide pool"> <img src="https://github.com/LotharukpongJS/LotharukpongJS.github.io/blob/main/images/PXL_20230823_163653386.MP.jpg?raw=true" width="200" alt="Ectocarpus.">

---

## Publications

> Barrera-Redondo, J., **Lotharukpong, J. S.**, Drost, H-G., & Coelho, S. M. (2023). Uncovering gene-family founder events during major evolutionary transitions in animals, plants and fungi using GenEra. [Genome Biology, 24(1), 54.](https://doi.org/10.1186/s13059-023-02895-z)
>  
> Manley, B. F., **Lotharukpong, J. S.**, Barrera-Redondo, J., Llewellyn, T., Yildirir, G., Sperschneider, J., ... & Dallaire, A. (2023). A highly contiguous genome assembly reveals sources of genomic novelty in the symbiotic fungus Rhizophagus irregularis. [G3: Genes, Genomes, Genetics, 13(6), jkad077.](https://doi.org/10.1093/g3journal/jkad077)
>
> **Lotharukpong, J. S.**, Zheng, M., Luthringer, R., Liesner, D., Drost, H-G., & Coelho, S. M. (2024). A transcriptomic hourglass in brown algae. [Nature 635, 129–135](https://doi.org/10.1038/s41586-024-08059-8)
>

See also on [google scholar](https://scholar.google.com/citations?user=2HiLuNEAAAAJ&hl)

## Media

The Node
> [Behind the paper: A transcriptomic hourglass in brown algae](https://thenode.biologists.com/behind-the-paper-a-transcriptomic-hourglass-in-brown-algae/research/)
>
European Reference Genome Atlas (ERGA)
> [Evolutionary transcriptomics in brown algae](https://www.erga-biodiversity.eu/post/a-transcriptomic-hourglass-in-brown-algae)

